homeassistant/components/sensor/mhz19.py

```diff
@@ -37,50 +52,101 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         _LOGGER.error("Could not open serial connection to %s (%s)",
                       config.get(CONF_SERIAL_DEVICE), err)
         return False
+    SENSOR_TYPES[SENSOR_TEMPERATURE][1] = hass.config.units.temperature_unit
+
+    data = MHZClient(co2sensor, config.get(CONF_SERIAL_DEVICE))
+    dev = []
+    name = config.get(CONF_NAME)
 
-    dev = MHZ19Sensor(config.get(CONF_SERIAL_DEVICE), config.get(CONF_NAME))
-    add_devices([dev])
+    for variable in config[CONF_MONITORED_CONDITIONS]:
+        dev.append(
+            MHZ19Sensor(data, variable, SENSOR_TYPES[variable][1], name))
+
+    add_devices(dev)
+    return True
 
 
 class MHZ19Sensor(Entity):
     """Representation of an CO2 sensor."""
 
-    def __init__(self, serial_device, name):
+    def __init__(self, mhz_client, sensor_type, temp_unit, name):
         """Initialize a new PM sensor."""
+        self._mhz_client = mhz_client
+        self._sensor_type = sensor_type
+        self._temp_unit = temp_unit
         self._name = name
-        self._state = None
-        self._serial = serial_device
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._ppm = None
+        self._temperature = None
+        self.update()
```

Comment:
Please remove update and instead pass `True` as second parameter to `add_devices`: `add_devices(dev, True)`